### ISSUE TITLE
Crash when adding a ASEditableTextNode to a horizontal stack layout spec

### ISSUE THREAD

**Issue by flovouin** [facebook/AsyncDisplayKit#1701]
Hi guys,

Following a discussion on Slack, I'm posting this issue concerning the ASEditableTextNode and stack layouts. The problem is simple: the `ASEditableTextNode` seems to return a size with an infinite width by default, and the horizontal stack layout has troubles handling this case. In fact, it will make the layout crash at runtime.

Since the discussion on Slack I found the solution: setting both `flexGrow` and `flexShrink` to `true`. As the `ASEditableTextNode` returns an infinite width, using `flexShrink` kind of makes sense. However `flexGrow` is also required otherwise the crash will occur.
Also, this behaviour is not very intuitive as it implies that the user should know that the `ASEditableTextNode` returns an infinite width by default, which seems more of an internal matter. (See the vertical stack example in the code linked below, which works as expected without ever specifying a size for the `ASEditableTextNode`.)
Another (dirtier) solution is to put the node inside a `ASStaticLayoutSpec` and set its `preferredFrameSize`, but that implies computing its size manually, which defeats the purpose of the layout engine.

More details about the crash and the solution are available in this example :
https://github.com/flovouin/ASEditableTextNode-Layout

Bottom line: given the behaviour of other nodes in the framework, I would expect the code that currently crashes ASDK to work (i.e. only setting `flexGrow` to `true` should make the `ASEditableTextNode` fill the available space).

Please let me know if you need any additional details :)

Flo

**Comment by WildDylan** [comment]
Hi, unfortunately you can’t position that in a layout spec if you are relying on the size of the element, because it is a UITextView

we can improve this warning / try to improve the default behavior

it should work if you make an ASStaticLayoutSpec and apply a sizeRange = ASSizeRangeMakeWithExactCGSize.  put the editable text node inside the static spec

**Comment by WildDylan** [comment]
ASStackLayoutSpec * spec = [ASStackLayoutSpec horizontalStackLayoutSpec];
    spec.spacing = 15;
    spec.verticalAlignment = ASVerticalAlignmentCenter;
    
    NSMutableArray * childArray = [NSMutableArray array];
    
    ASStaticLayoutSpec * inputLayout = [ASStaticLayoutSpec staticLayoutSpecWithChildren:@[self.inputText]];
    inputText.sizeRange = ASRelativeSizeRangeMakeWithExactCGSize(CGSizeMake(100, constrainedSize.min.width - 80));
    
    [childArray addObjectsFromArray:@[self.titleNode, self.vert_sepLine, inputLayout]];

**Comment by flovouin** [comment]
Yes, it's the "dirty solution" I mentioned and this works, but defeats the purpose of the stack layout.
See the `flexGrow` and `flexShrink` trick to get a nicer solution. Still, I wonder if this is the expected behaviour, and don't find it very intuitive.

**Comment by appleguy** [comment]
We definitely have an issue to resolve here.  The unfortunate situation currently, which has not been realized before because this class is fairly rare to use, is that editable text node supports only manual layout or being contained in a static layout spec.  It will not work with automatic sizing.  You could get around this by using an ASTextNode in its place in layoutSpecThatFits, perhaps marked hidden, with the same text -- and setting the editable node to the same frame in the -layout method (layout is different than layoutSpecThatFits)

**Comment by maicki** [comment]
Hi all, I put up a PR that should fix the crash and the behavior of a `ASEditableTextNode` in a horizontal stack. In terms of calculate the intrinsic content size, it's is working now the same way as a `ASTextNode` .

Furthermore added support for setting `preferredFrameSize`.

Let me know if this resolve you problems ...

**Comment by flovouin** [comment]
Thanks appleguy for the reply, and thanks maicki for the PR! I can confirm that everything works perfectly  in the example linked above. I haven't tested in my proper project, but there's no reason it would behave differently.
Thank you so much for fixing this so quickly!
Closed. :)
